### PR TITLE
Feature/router park info

### PR DESCRIPTION
**Pull request by jaypeasee**
## Is this a fix or a feature?
Feature

## What is the change?
1. Sets up Router for the entire app - `/:parkCode`, `/about` etc.
2. Gets a random park appear only when the route is `/`
3. Creates a controlled form on the MainNav

## Where should the reviewer start?
1. Review the return statement in App.
2. Review MainNav so see how our filter and controlled form now works.

## How should this be tested?
* Load the page, see that a random park loads when the path is `/`. 
* Click on a park to see the route change. Refresh to confirm you stay on that park and the route doesn't change
* Search for park buttons in the controlled form.

## Relevant Tickets
1. #34 
2. #22 
3. #23 

@saraho1123
@jaypeasee
@baileydunning

### CODE REVIEW

File: src/App/App.tsx [hunk 1]
```diff
@@ -2,40 +2,43 @@ import './App.scss'
...
-}
-
-interface Park {
-  parkCode: string;
-}
-
 const App: React.FC = () => {
   const [parkCode, setParkCode] = useState('')
-  
+  const location = useLocation()
```
Comment (saraho1123) [inline src/App/App.tsx:11]: This was so smart Bailey! Way to go!

File: src/App/App.tsx [hunk 2]
```diff
@@ -2,40 +2,43 @@ import './App.scss'
...
+      <Switch>
+        <Route
+          path={location.pathname}
+          render={ () => {
+            return (
+              <ParkContainer parkCode={parkCode}/>
+            )
+          }}
+        />
+      </Switch>
```
Comment (saraho1123) [inline src/App/App.tsx:41]: Were we going to add a comment here to come back and add an Error component to this switch?

**Comment by baileydunning** [comment]
Looks great team! We made a ton of progress with this branch 😎

**Review by baileydunning — APPROVED** [review]

File: src/MainNav/MainNav.tsx [hunk 3]
```diff
@@ -1,14 +1,57 @@
...
+                choosePark={choosePark}
+            />
+        })
+        setParksOnDisplay(parkButtons)
+    }
+
+    const filterButtonsByName = (searchTerm: string) => {
+        setNameSearch(searchTerm)
+            const filteredParks = nationalParks.filter(park => {
+            return park.name.toLowerCase().includes(searchTerm.toLowerCase())
```
Comment (saraho1123) [inline src/MainNav/MainNav.tsx:50]: Way to go thinking of toLowerCase, JP!

File: src/NavSearch/NavSearch.tsx [hunk 4]
```diff
@@ -1,14 +1,29 @@
...
 
-function NavSearch() {
+interface FilterButtonsByName {
+    filterButtonsByName: (searchTerm: string) => void
+}
+
+const NavSearch: React.FC<FilterButtonsByName> = (props) => {
+    const [nameSearch, setNameSearch] = useState<string>('')
+
+    const handleSearchChange = (event: any) => {
```
Comment (saraho1123) [inline src/NavSearch/NavSearch.tsx:12]: Did we want to come back and fix this any type?

**Review by saraho1123 — APPROVED** [review]
typeScript is a pain ... but we are doing a great job learning this!

**Pull request merged**